source/extensions/quic_listeners/quiche/platform/quic_mem_slice_span_impl.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "envoy/buffer/buffer.h"
+
+#include "common/common/stack_array.h"
+
+#include "quiche/quic/core/quic_types.h"
+#include "quiche/quic/platform/api/quic_mem_slice.h"
+#include "quiche/quic/platform/api/quic_string_piece.h"
+
+namespace quic {
+
+// Wraps a Buffer::Instance and deliver its data with mininum number of copies.
+class QuicMemSliceSpanImpl {
+public:
+  QuicMemSliceSpanImpl() : buffer_(nullptr) {}
+  /**
+   * @param buffer has to outlive the life time of this class.
+   */
+  explicit QuicMemSliceSpanImpl(Envoy::Buffer::Instance& buffer) : buffer_(&buffer) {}
+
+  QuicMemSliceSpanImpl(const QuicMemSliceSpanImpl& other) = default;
+  QuicMemSliceSpanImpl& operator=(const QuicMemSliceSpanImpl& other) = default;
+
+  QuicMemSliceSpanImpl(QuicMemSliceSpanImpl&& other) noexcept : buffer_(other.buffer_) {
+    other.buffer_ = nullptr;
+  }
+
+  QuicMemSliceSpanImpl& operator=(QuicMemSliceSpanImpl&& other) noexcept {
+    if (this != &other) {
+      buffer_ = other.buffer_;
+      other.buffer_ = nullptr;
+    }
+    return *this;
+  }
+
+  QuicStringPiece GetData(size_t index) {
+    uint64_t num_slices = buffer_->getRawSlices(nullptr, 0);
+    ASSERT(num_slices > index);
+    Envoy::STACK_ARRAY(slices, Envoy::Buffer::RawSlice, num_slices);
+    buffer_->getRawSlices(slices.begin(), num_slices);
+    return {reinterpret_cast<char*>(slices[index].mem_), slices[index].len_};
+  }
+
+  QuicByteCount total_length() { return buffer_->length(); };
```

Comment:
got it. again since this isn't actually a subclass can you put a comment above the 'class' declaration pointing to the interface this is supposed to implement?

Reply:
done

Reply:
Doesn't look done?

Reply:
thanks :)